charmcraft/logsetup.py

```diff
@@ -103,17 +104,25 @@ def _set_filehandler(self):
         self._file_logger.addHandler(file_handler)
         self._file_logger.debug("Starting charmcraft version %s", __version__)
 
+    def _cleanup_log(self):
+        if self._log_fd is not None:
+            os.close(self._log_fd)
+        self._log_fd = None
```

Comment:
Why not telling the whole logging system to release the file handlers? It can be done through [shutdown](https://docs.python.org/3/library/logging.html#logging.shutdown).

Reply:
The handle is opened with mkstemp() and we need to close it.  Anyways, isn't this all going to be replaced in the near future with craft-cli? :D

Reply:
Probably we should close the file descriptor opened by mkstemp anyway, the moment after creating the temp file, not storing to the future. In any case, even if we close the filedescriptor (after creating the temp file, or at the end), we should tell `logging` to shutdown, right? otherwise we won't be able to unlink the file anyway....